src/subdomains/generic/user/models/user-data/user-data.service.ts

```diff
@@ -165,6 +165,15 @@ export class UserDataService {
       if (identCompleted && pendingVideo) userData.cancelStep(pendingVideo);
     }
 
+    // If KYC level >= 50 and DFX-approval not complete, complete it.
+    if (userData.kycLevel >= KycLevel.LEVEL_50) {
+      const dfxApprovalCompleted = userData.hasCompletedStep(KycStepName.DFX_APPROVAL);
+      if (!dfxApprovalCompleted) {
```

Comment:
Do we have to check if there are completedSteps of DfxApproval?
If we have Pending DfxApproval steps I would just complete these or do we think that users will have more than one dfxApproval step? 🤔